src/call_log/call_log.c

```diff
@@ -262,6 +262,35 @@ __wt_call_log_query_timestamp(
     return (0);
 }
 
+int
+__wt_call_log_session_query_timestamp(
```

Comment:
The only differences between `__wt_call_log_conn_query_timestamp` and `__wt_call_log_session_query_timestamp` are minimal, I can only see:

```
WT_RET(__call_log_print_start(session, "connection", "query_timestamp"));
WT_RET(__call_log_print_start(session, "session", "query_timestamp"));
```
And
```
WT_RET(__wt_fprintf(session, conn->call_log_fst, "    \"connection_id\": \"%p\",\n", conn));
WT_RET(__wt_fprintf(session, conn->call_log_fst, "    \"session_id\": \"%p\",\n", session));
```
Unless I am missing other differences, we should be able to avoid duplication.

Reply:
Yeah that's true, I can combine the two.

Reply:
I've combined the functions and added a bool `global` to indicate whether we're querying the global transaction timestamps on the connection or the session transaction timestamps.

Reply:
It looks better! I think we can push it further by simply having two variables that would be set to `connection` and `connection_id` or `session` and `session_id` and call the two functions with those variables correctly set depending on the value of `global`.

Reply:
As discussed, it will lower the readability fo the code if we do what I suggested. Sorry!